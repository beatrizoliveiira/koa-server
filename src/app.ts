const koa = require('koa');
const graphqlHTTP = require('koa-graphql');
const Router = require('@koa/router');
const schema = require('./graphql/schema').default;
const root = require('./graphql/root');

const app = new koa();
const router = new Router();

app.on('error', err => {
  console.log('Server error', err);
});

router.get( '/',  async  (ctx) => {
    ctx.body = 'Hello World'
})

router.all('/graphql', graphqlHTTP({
  schema,
  graphiql: true,
  rootValue: root,
}));

app.use(router.routes()).use(router.allowedMethods());

export default app;