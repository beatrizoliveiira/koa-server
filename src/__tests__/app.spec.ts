import request from 'supertest';
import { clearDbAndRestartCounters, connectMongoose, disconnectMongoose } from '../../test';
import app from '../app';
import { createEvent } from '../modules/event/fixture/createEvent';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

it('should return 200', async () => {
  const event = await createEvent({
    name: 'event',
    start: '2019-01-01T00:00:00.000Z',
    end: '2019-01-01T23:59:59.000Z',
    allDay: true,
  });

  // language=GraphQL
  const query = `
    query Q {
      events {
        edges {
          node {
            name
            start
            end
            allDay
          }
        }
      }      
    }
  `;

  const variables = {};

  const payload = {
    query,
    variables,
  };

  const response = await request(app.callback())
    .post('/graphql')
    .set({
      Accept: 'application/json',
      'Content-Type': 'application/json',
    })
    .send(JSON.stringify(payload));

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(200);
});
