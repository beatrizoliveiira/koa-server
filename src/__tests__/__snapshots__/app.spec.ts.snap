// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return 200 1`] = `
Object {
  "data": Object {
    "events": Object {
      "edges": Array [
        Object {
          "node": Object {
            "allDay": true,
            "end": "2019-01-01T23:59:59.000Z",
            "name": "event",
            "start": "2019-01-01T00:00:00.000Z",
          },
        },
      ],
    },
  },
}
`;
